File: src/main/java/hu/frontrider/gearcraft/proxy/ClientProxy.java
package hu.frontrider.gearcraft.proxy;

import hu.frontrider.gearcraft.client.renderer.MechGolemRenderer;
import hu.frontrider.gearcraft.entities.EntityMechanicalGolem;
import hu.frontrider.gearcraft.registry.ItemRegistry;
import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.ItemModelMesher;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.item.Item;
import net.minecraftforge.client.model.ModelLoader;
import net.minecraftforge.fml.client.registry.RenderingRegistry;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

import java.util.Objects;

public class ClientProxy extends CommonProxy {

    @Override
    public void init(FMLInitializationEvent e) {
        super.init(e);
        initClient(Minecraft.getMinecraft().getRenderItem().getItemModelMesher());
        RenderingRegistry.registerEntityRenderingHandler(EntityMechanicalGolem.class, MechGolemRenderer::new);
    }

    @SideOnly(Side.CLIENT)
    public static void initClient(ItemModelMesher mesher) {
        for (Item item : ItemRegistry.items) {
            ModelResourceLocation model = new ModelResourceLocation(Objects.requireNonNull(item.getRegistryName()).toString(), "inventory");
            ModelLoader.registerItemVariants(item, model);
            mesher.register(item, 0, model);
        }
    }
}

File: mod/src/main/java/hu/frontrider/gearcraft/client/ClientProxy.kt
package hu.frontrider.gearcraft.client

import hu.frontrider.gearcraft.basePlugin
import hu.frontrider.gearcraft.items.ColoredItem
import hu.frontrider.gearcraft.proxy.CommonProxy
import hu.frontrider.gearcraft.core.registry.ItemRegistry
import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.ItemModelMesher
import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraft.client.renderer.color.IItemColor
import net.minecraft.util.ResourceLocation
import net.minecraftforge.client.model.ModelLoader
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import java.util.*

class ClientProxy : CommonProxy() {

    override fun init(e: FMLInitializationEvent) {
        super.init(e)
        initClient(Minecraft.getMinecraft().renderItem.itemModelMesher)
    }

    @SideOnly(Side.CLIENT)
    fun initClient(mesher: ItemModelMesher) {
        for (item in ItemRegistry.items) {
            val model = ModelResourceLocation(Objects.requireNonNull<ResourceLocation>(item.registryName).toString(), "inventory")
            ModelLoader.registerItemVariants(item, model)
            mesher.register(item, 0, model)
        }

        Minecraft.getMinecraft().itemColors.registerItemColorHandler(
                IItemColor { stack, tintIndex ->
                    (stack.item as ColoredItem).color[tintIndex]
                },
                *basePlugin.items.filter { it is ColoredItem }.toTypedArray()
        )

    }
}
